src/sys/signal.rs

```diff
@@ -6,44 +6,125 @@ use {Errno, Result};
 use std::mem;
 use std::ptr;
 
-pub use libc::{
-    SIGHUP,
-    SIGINT,
-    SIGQUIT,
-    SIGILL,
-    SIGABRT,
-    SIGFPE,
-    SIGKILL,
-    SIGSEGV,
-    SIGPIPE,
-    SIGALRM,
-    SIGTERM,
-    SIGTRAP,
-    SIGIOT,
-    SIGBUS,
-    SIGSYS,
-    SIGURG,
-    SIGSTOP,
-    SIGTSTP,
-    SIGCONT,
-    SIGCHLD,
-    SIGTTIN,
-    SIGTTOU,
-    SIGIO,
-    SIGXCPU,
-    SIGXFSZ,
-    SIGVTALRM,
-    SIGPROF,
-    SIGWINCH,
-    SIGUSR1,
-    SIGUSR2,
-};
-
-// This doesn't always exist, but when it does, it's 7
-pub const SIGEMT: libc::c_int = 7;
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum Signal {
+    SIGHUP = libc::SIGHUP as isize,
```

Comment:
I came across the same issue in #364 though I decided to go with `#[repr(i32)]` since the type of the constants in libc crate (`c_int` is `i32` unconditionally), and cast to `c_int` at use. Thinking about it a bit more, I'm not super happy about it, because if libc ever makes `c_int` conditional, we'll have trouble.